### ISSUE TITLE
2.3.0 Unable to import products with remote images (CatalogImportExport)

### ISSUE THREAD

**Issue by intd** [magento/magento2#21881]
### Preconditions (*)

1. Magento 2.3.0
2. PHP 7.2

### Steps to reproduce (*)

1. Import products (csv file with remote images)

### Expected result (*)

1. The product has been imported into the store

### Actual result (*)

1.Error:  Imported resource (image) could not be downloaded from external resource due to timeout or access permissions in row(s): 

### Note (*)

The import only works from a local folder ( e.g. var/import).

Image link:
https://example.com/image.jpg

exception.log
> main.CRITICAL: File 'pub/media/import/image.jpgimage.jpg' was not found or has read restriction.

The file image.jpg has been downloaded by Magento properly and  it's in the directory 'pub/media/import/image.jpg'

I was able to import the products by this workaround checking if the image comes from a local or remote

https://github.com/magento/magento2/blob/497e94ade4211891531923742f74992cedb0395b/app/code/Magento/CatalogImportExport/Model/Import/Uploader.php#L163

```
   $isUrl = false;
   if (preg_match('/\bhttps?:\/\//i', $fileName, $matches)) {
    $isUrl = true;
```

https://github.com/magento/magento2/blob/497e94ade4211891531923742f74992cedb0395b/app/code/Magento/CatalogImportExport/Model/Import/Uploader.php#L190

```
  $imagePath = $isUrl ? $filePath  : $filePath . $fileName;
  $filePath = $this->_directory->getRelativePath($imagePath);
```

**Comment by magento-engcom-team** [comment]
Hi @intd. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@intd do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by intd** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @intd. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @intd, here is your [Magento instance](https://i-21881-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-21881-2-3-develop.instances.magento-community.engineering/admin](https://i-21881-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by intd** [comment]
> Hi @intd. Thank you for your report.
> To help us process this issue please make sure that you provided the following information:
> 
>     * [*]  Summary of the issue
> 
>     * [*]  Information on your environment
> 
>     * [*]  Steps to reproduce
> 
>     * [*]  Expected and actual results
> 
> 
> Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue:
> 
> `@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release
> 
> For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.
> 
> @intd do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
> 
>     * [ ]  yes
> 
>     * [*]  no

There is no issue on the develop version

**Comment by magento-engcom-team** [comment]
Hi @intd. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @intd, here is your [Magento instance](https://i-21881-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-21881-2-3-develop.instances.magento-community.engineering/admin](https://i-21881-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by intd** [comment]
issue is not reproducible

**Comment by ghost** [comment]
@intd yes this was fixed in 2.3-develop, you closed fastly than i can find commit. :) 
https://github.com/magento/magento2/pull/20127
